spt/features/playerio.cpp

```diff
@@ -512,25 +501,84 @@ void PlayerIOFeature::OnTick()
 	currentVelocity = GetPlayerVelocity();
 }
 
+static const std::map<std::string, int> buttonCodeMap = {
+    {"attack", 1 << 0},
+    {"jump", 1 << 1},
+    {"duck", 1 << 2},
+    {"forward", 1 << 3},
+    {"back", 1 << 4},
+    {"use", 1 << 5},
+    {"left", 1 << 7},
+    {"right", 1 << 8},
+    {"moveleft", 1 << 9},
+    {"moveright", 1 << 10},
+    {"attack2", 1 << 11},
+    {"reload", 1 << 13},
+    {"speed", 1 << 17},
+    {"walk", 1 << 18},
+    {"zoom", 1 << 19},
+};
+
 #if defined(OE)
-static void DuckspamDown()
+static void SpamDown_wrapper(ArgsWrapper args);
+static void SpamDown()
+{
+	if (!interfaces::engine)
+		return;
+	ArgsWrapper args;
+	SpamDown_wrapper(args);
+}
+static void SpamDown_wrapper(ArgsWrapper args)
 #else
-static void DuckspamDown(const CCommand& args)
+static void SpamDown(const CCommand& args)
 #endif
 {
-	spt_playerio.EnableDuckspam();
+	if (args.ArgC() < 2)
+	{
+		Msg("Usage: +y_spt_spam <key>\n"
+		    "keys: attack, jump, duck, forward, back, use, left, right, moveleft, moveright, attack2, reload, speed, walk, zoom\n");
+		return;
+	}
+
+	auto buttonCodeIter = buttonCodeMap.find(args.Arg(1));
+	if (buttonCodeIter != buttonCodeMap.end())
+		spt_playerio.EnableSpam(buttonCodeIter->second);
+	else
+		Msg("Cannot find key <%s>\n", args.Arg(1));
 }
-static ConCommand DuckspamDown_Command("+y_spt_duckspam", DuckspamDown, "Enables the duckspam.");
+static ConCommand SpamDown_Command(
+    "+y_spt_spam",
+    SpamDown,
+    "Enables key spam.\n"
+    "Usage: +y_spt_spam <key>\n"
+    "keys: attack, jump, duck, forward, back, use, left, right, moveleft, moveright, attack2, reload, speed, walk, zoom");
 
 #if defined(OE)
-static void DuckspamUp()
+static void SpamUp_wrapper(ArgsWrapper args);
+static void SpamUp()
+{
+	if (!interfaces::engine)
+		return;
+	ArgsWrapper args;
+	SpamUp_wrapper(args);
+}
+static void SpamUp_wrapper(ArgsWrapper args)
 #else
-static void DuckspamUp(const CCommand& args)
+static void SpamUp(const CCommand& args)
```

Comment:
the ifdef spam here looks a bit dirty, feel like this could be replaced with a macro